cunumeric/array.py

```diff
@@ -4168,8 +4170,17 @@ def unique(self) -> ndarray:
         Multiple GPUs, Multiple CPUs
 
         """
-        thunk = self._thunk.unique()
-        return ndarray(shape=thunk.shape, thunk=thunk)
+        thunk = self._thunk.unique(return_index)
```

Comment:
[nit] -- i wouldn't call this `thunk` anymore, since when `return_index=True` it's not a thunk anymore, it's a tuple of thunks.